imagebuildah/stage_executor_test.go

```diff
@@ -0,0 +1,90 @@
+package imagebuildah
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHistoryEntriesEqual(t *testing.T) {
+	testCases := []struct {
+		a, b  string
+		equal bool
+	}{
+		{
+			a:     `{}`,
+			b:     `{}`,
+			equal: true,
+		},
+		{
+			a:     `{"created":"2020-06-17T00:22:25.47282687Z"}`,
+			b:     `{"created":"2020-06-17T00:22:25.47282687Z"}`,
+			equal: true,
+		},
+		{
+			a:     `{"created":"2020-07-16T12:38:26.733333497-04:00"}`,
+			b:     `{"created":"2020-07-16T12:38:26.733333497-04:00"}`,
+			equal: true,
+		},
+		{
+			a:     `{"created":"2020-07-16T12:38:26.733333497-04:00"}`,
+			b:     `{"created":"2020-07-16T12:38:26.733333497Z"}`,
+			equal: false,
+		},
+		{
+			a:     `{"created":"2020-07-16T12:38:26.733333497Z"}`,
+			b:     `{}`,
+			equal: false,
+		},
+		{
+			a:     `{}`,
+			b:     `{"created":"2020-07-16T12:38:26.733333497Z"}`,
+			equal: false,
+		},
+		{
+			a:     `{"comment":"thing"}`,
+			b:     `{"comment":"thing"}`,
+			equal: true,
+		},
+		{
+			a:     `{"comment":"thing"}`,
+			b:     `{"comment":"things"}`,
+			equal: false,
+		},
+		{
+			a:     `{"author":"respected"}`,
+			b:     `{"author":"respected"}`,
+			equal: true,
+		},
+		{
+			a:     `{"author":"respected"}`,
+			b:     `{"author":"discredited"}`,
+			equal: false,
+		},
+		{
+			a:     `{"created_by":"actions"}`,
+			b:     `{"created_by":"actions"}`,
+			equal: true,
+		},
+		{
+			a:     `{"created_by":"jiggery"}`,
+			b:     `{"created_by":"pokery"}`,
+			equal: false,
+		},
+	}
```

Comment:
Is casing significant?  If so, might want to add a test or two.

Reply:
`encoding/json`'s godoc for `Unmarshal()` says it prefers an exact match, but will accept a case-insensitive one.

Reply:
thanks!  I thought that might be the case, but I wasn't sure.  I appreciate you running that down.